polar-core/src/partial/constraints.rs

```diff
@@ -520,6 +521,44 @@ mod test {
         Ok(())
     }
 
+    #[test]
+    fn test_partial_inverter() -> TestResult {
```

Comment:
What about some of the tests I proposed? In particular when the not is not the first thing that happens to that partial.  Also when lookups are involved (I'm concerned the constraint merging will not happen correctly because at this point all the temporary variables that represent the same value (result of `x.foo`) will have different names).

I think there are some bugs we need to fix.

Reply:
Also nested nots like:

```
if not (not (not x = 1));
if not (x = 1 AND (not x = 2));
if x = 3 AND not (x = 1 AND (not x = 2));
```

stuff like this, combined with ors, etc.

Reply:
Tests added in 8257779 and fixed in 8257779 + 68db0ad